### ISSUE TITLE
Incorrect handling of default type parameter values produces infinite changes

### ISSUE THREAD

**Issue by arthurgubaidullin** [hosseinmd/prettier-plugin-jsdoc#176]
Example of buggy code:
```js
/**
 * @template [BAR=({bar:true}&{foo:false}&{buggy:true})] 
 * @typedef {{foo: BAR}} SomeType
 */
```
First formatting produces the following result:
```js
/**
 * @template [BAR=({bar:true}&{foo:false}&{buggy:true})] Default is
 *   `({bar:true}&{foo:false}&{buggy:true})`
 * @typedef {{foo: BAR}} SomeType
 */
```
Second formatting produces the following result:
```js
/**
 * @template [BAR=({bar:true}&{foo:false}&{buggy:true})] Default is
 *   `({bar:true}&{foo:false}&{buggy:true})`. Default is
 *   `({bar:true}&{foo:false}&{buggy:true})`
 * @typedef {{foo: BAR}} SomeType
 */
```
Third formatting produces the following result:
```js
/**
 * @template [BAR=({bar:true}&{foo:false}&{buggy:true})] Default is
 *   `({bar:true}&{foo:false}&{buggy:true})`. Default is
 *   `({bar:true}&{foo:false}&{buggy:true})`. Default is
 *   `({bar:true}&{foo:false}&{buggy:true})`
 * @typedef {{foo: BAR}} SomeType
 */
```
And so on.

Prettier config:
```json
{
  "bracketSpacing": false,
  "singleQuote": true,
  "plugins": ["./node_modules/prettier-plugin-jsdoc"]
}
```

Environment:
- prettier-plugin-jsdoc@0.4.2
- prettier@2.7.1

**Comment by arthurgubaidullin** [comment]
I think the problem is on this line. A possible line break is not checked.

https://github.com/hosseinmd/prettier-plugin-jsdoc/blob/d255d76b93aba6bb528c0db34e2de7e4a6e7ce57/src/parser.ts#L533

**Comment by levithomason** [comment]
I'd like to disable this feature altogether. It is unexpected for a formatter to write documentation for me. I don't want my jsdoc content changed, I only want it formatted.

The jsdoc in that file says that it is in place due to a TS issue, however, I'm not using TS and my IDE also shows the defaults already. So, the insertion is unnecessary. It is cluttering up the actual documentation in my case:

<img width="648" alt="image" src="https://user-images.githubusercontent.com/5067638/209427761-7ad811c0-c4e5-418c-8eb8-faa49efacd73.png">

<img width="635" alt="image" src="https://user-images.githubusercontent.com/5067638/209427801-468b4e41-0ad6-4814-8166-c0884dbd3cea.png">

**Comment by hosseinmd** [comment]
Yes, you are right. This feature is for VS Code. I agree with you, this feature should be optional.

**Comment by wreckah** [comment]
Just want to clarify that such feature is [already implemented](https://github.com/hosseinmd/prettier-plugin-jsdoc/commit/c3fb1680d9fd20678d4fb4069395e14fc0440e19), but is not referred in the docs:
```js
module.exports = {
  plugins: ['prettier-plugin-jsdoc'],
  jsdocAddDefaultToDescription: false
};

**Comment by hosseinmd** [comment]
fixed: v1.0.5
